tests/Services/LogWriterTest.php

```diff
@@ -39,10 +38,38 @@ public function testListByLineStatusSuccessfully(): void
      * Test write log successfully.
      *
      * @return void
+     *
+     * @throws \Exception
      */
     public function testWriteSuccessfully(): void
     {
-        $this->getService()->write(new DtoStub());
+        $this->getService()->write(new LogLine(
+            '127.0.01',
+            1,
+            new \DateTime(),
+            '{"send": "me"}',
+            '{"status": "ok"}'
+        ));
+
+        $this->addToAssertionCount(1);
```

Comment:
A better test to is replace these addToAssertionCount with comparisons with an expected Result object.

This means if service being tested stops returning results, the test will go red. (which is what we want)

Reply:
In this case the service being tested throws exception i.e. either it is successful or throws exception.

Reply:
Ok, so if it does not, what's the behaviour of LogWriter we want to verify? In this case, it's going to be that the expected to Logline object was written to the underlying service.

If this is refactored into the Arrange, Act, Assert pattern and declare the `new Logline()` in it's own variable, then it's pretty trivial to compare the variable against what is written into `LogWriterStub`.

Reply:
Solved. DocumentManagerStub now manages data written and updated which in turn helps test  assert if data was actually written or updated.